pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1161,6 +1161,50 @@ func TestReconcileOnCancelledPipelineRun(t *testing.T) {
 	}
 }
 
+func TestReconcileOnPendingPipelineRun(t *testing.T) {
+	// TestReconcileOnPendingPipelineRun runs "Reconcile" on a PipelineRun that is pending.
+	// It verifies that reconcile is successful, the pipeline status updated and events generated.
+	prs := []*v1beta1.PipelineRun{tb.PipelineRun("test-pipeline-run-pending",
+		tb.PipelineRunNamespace("foo"),
+		tb.PipelineRunSpec("test-pipeline", tb.PipelineRunServiceAccountName("test-sa"),
+			tb.PipelineRunPending,
+		),
+		tb.PipelineRunStatus(tb.PipelineRunStartTime(time.Now())),
+	)}
+	ps := []*v1beta1.Pipeline{tb.Pipeline("test-pipeline", tb.PipelineNamespace("foo"), tb.PipelineSpec(
+		tb.PipelineTask("hello-world", "hello-world"),
+	))}
+	ts := []*v1beta1.Task{tb.Task("hello-world", tb.TaskNamespace("foo"))}
+	trs := []*v1beta1.TaskRun{
+		tb.TaskRun("test-pipeline-run-pending-hello-world",
+			tb.TaskRunNamespace("foo"),
+			tb.TaskRunOwnerReference("kind", "name"),
+			tb.TaskRunLabel(pipeline.GroupName+pipeline.PipelineLabelKey, "test-pipeline-run-pending"),
+			tb.TaskRunLabel(pipeline.GroupName+pipeline.PipelineRunLabelKey, "test-pipeline"),
+			tb.TaskRunSpec(tb.TaskRunTaskRef("hello-world"),
+				tb.TaskRunServiceAccountName("test-sa"),
+			),
+		),
+	}
+
+	d := test.Data{
+		PipelineRuns: prs,
+		Pipelines:    ps,
+		Tasks:        ts,
+		TaskRuns:     trs,
```

Comment:
im a bit confused by why the task run exists? if the pipelinerun is created "pending" then i think no taskruns should exist? i.e. there should never be a time when a pending pipelinerun exists and taskruns exist for the same pipelinerun?

Reply:
Yeah, actually I brought this over from the pause PR that I based this on and just trusted it, but now I'm realizing I don't understand it at all. I'll work on it :)

Reply:
Okay, it should be more correct now.